Lib/typing.py

```diff
@@ -3783,6 +3784,44 @@ def get_protocol_members(tp: type, /) -> frozenset[str]:
     return frozenset(tp.__protocol_attrs__)
 
 
+_P = ParamSpec("_P")
+
+
+def copy_kwargs(
+    source_func: Callable[_P, Any]
+) -> Callable[[Callable[..., T]], Callable[_P, T]]:
+    """Cast the decorated function's call signature to the source_func's.
+
+    Use this decorator enhancing an upstream function while keeping its
+    call signature.
+    Returns the original function with the source_func's call signature.
+
+    Usage::
+
+        from typing import copy_kwargs, Any
+
+        def upstream_func(a: int, b: float, *, double: bool = False) -> float:
+            ...
+
+        @copy_kwargs(upstream_func)
```

Comment:
The name `copy_kwargs` implies that it does something to only the kwargs and not the args, but that's not the case; it copies both.

I would use `@copy_signature`.

Reply:
Hi, 

sorry haven't had time to work on this in the meanwhile.
I agree that `copy_kwargs` is not correct but neither is `copy_signature`. As in my understanding a signature includes also the return value. And we explicitly do not copy these.

After some thinking I came up with:
`copy_func_params`.

This also opens up the possibility to create a `copy_meth_params`.
The `copy_meth_params` is the same as `copy_func_params` but keeps the `self` (or `cls`) argument.